File: Port.java
import java.awt.Color;

import java.awt.Graphics;

import java.util.ArrayList;

public class Port {

	ArrayList<ClassArray<ITechno>> portStages;

	int countPlaces = 20;

	int placeWidth = 210;

	int placeHeight = 80;

	int currentLevel;

	public Port(int countStages)

	{

		portStages = new ArrayList<ClassArray<ITechno>>(countStages);

		for (int i = 0; i < countStages; i++)

		{

			portStages.add(new ClassArray<ITechno>(countPlaces, null));

		}

	}

	public int getCurrentLevel() {

		return currentLevel;

	}

	public void levelUp()

	{

		if (currentLevel + 1 < portStages.size())
			currentLevel++;

	}

	public void levelDown()

	{

		if (currentLevel > 0)
			currentLevel--;

	}

	public int putPlaneInPort(ITechno plane)

	{

		return portStages.get(currentLevel).plus(portStages.get(currentLevel),
				plane);

	}

	public ITechno getPlaneInPort(int index)

	{

		return portStages.get(currentLevel).minus(portStages.get(currentLevel),
				index);

	}

	public void draw(Graphics g, int width, int height)

	{

		drawMarking(g);

		for (int i = 0; i < countPlaces; i++)

		{

			ITechno plane = portStages.get(currentLevel).getPlane(i);

			if (plane != null)

			{

				plane.setPosition(5 + i / 5 * placeWidth + 45, i % 5
						* placeHeight + 65);

				plane.draw(g);

			}

		}

	}

	public void drawMarking(Graphics g)

	{

		g.setColor(Color.BLACK);

		g.drawRect(0, 0, (countPlaces / 5) * placeWidth, 450);

		for (int i = 0; i < countPlaces / 5; i++)

		{

			for (int j = 0; j < 6; j++)

			{

				g.drawLine(i * placeWidth, j * placeHeight, i * placeWidth
						+ 110, j * placeHeight);

			}

			g.drawLine(i * placeWidth, 0, i * placeWidth, 400);

		}

	}

}

File: Port.cs
﻿using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WindowsFormsApp3
{
    class Port
    {
        ClassArray<Techno> port;
        int countPlaces = 20;
        int placeSizeWidth = 350;
        int placeSizeHeight = 80;
        public Port()
        {
            port = new ClassArray<Techno>(countPlaces, null);
        }
        public int PutShipInPort(Techno ship)
        {
            return port + ship;
        }
        public Techno GetShipInPort(int ticket)
        {
            return port - ticket;
        }
        public void Draw(Graphics g, int width, int height)
        {
            DrawMarking(g);
            for (int i = 0; i < countPlaces; i++)
            {
                var ship = port.getObject(i);
                if (ship != null)
                {
                    ship.setPosition(5 + i / 5 * placeSizeWidth + 45, i % 5 * placeSizeHeight + 65);
                    ship.draw(g);
                }
            }
        }
        private void DrawMarking(Graphics g)
        {
            Pen pen = new Pen(Color.Black, 3);
            g.DrawRectangle(pen, 0, 0, (countPlaces / 5) * placeSizeWidth, 480);
            for (int i = 0; i < countPlaces / 5; i++)
            {
                for (int j = 0; j < 6; ++j)
                {
                    g.DrawLine(pen, i * placeSizeWidth, j * placeSizeHeight,
                        i * placeSizeWidth + 110, j * placeSizeHeight);
                }
                g.DrawLine(pen, i * placeSizeWidth, 0,
                        i * placeSizeWidth, 400);
            }
        }
    }
}
